rfcs/051-concepts-adapters/README.md

```diff
@@ -0,0 +1,55 @@
+# RFC 051: Ingesting Library of Congress concepts
+
+As per the high-level design of [RFC 052](https://github.com/wellcomecollection/docs/pull/83), one of the key areas of the concepts pipeline topology is an ingest section. This RFC outlines how this will be implemented for the first work phase (attaching only LoC concepts to works).
+
+![High level outline of the architecture in this RFC](https://user-images.githubusercontent.com/4429247/177178393-8fd615f1-f178-4a0a-83b2-a1d5ee09634f.png)
+
+### Retrieving concepts from bulk exports
+LoC provide bulk exports of all [subject headings](https://id.loc.gov/authorities/subjects.html) (LCSH) and [names](https://id.loc.gov/authorities/names.html) (LCNAF). These come in variety of flavours: in both MADS and SKOS vocabularies; and in N-Triple, JSON-LD, Turtle, and XML formats. All of the bulk exports are gzipped.
+
+We can re-run the ingest process on a schedule to capture the (infrequent) changes to LoC concepts, which are almost always additions rather than changes/deletions. 
+
+In the architecture diagram above, the "adapter" and "transformer" are outlined as separate services. In practice, we can perform this mini-ETL operation more efficiently and simply as a streaming operation within one service when we use the gzipped JSON-LD files; these express "one concept per line" and so we needn't worry about traversing any trees or indeed caring much about the underlying RDF.
+
+![A more detailed view of how a concepts ingestor might work](https://user-images.githubusercontent.com/4429247/177571629-56f7dd7b-9c43-4f38-bb11-aae53ed0c2e6.png)
+
+The "transform" step here is (a) the most ill-defined and (b) the only step that needs to differ for LCSH and LCNAF ingestors.
+
+### Transforming external concepts 
+
+[RFC 050](https://github.com/wellcomecollection/docs/tree/main/rfcs/050-concepts-api) outlined the rough design of a concepts API: obviously, we need to store the external concepts in a way that is sufficient to construct these API documents. We also don't want to extraneous data, as the majority of the concepts won't be used in the catalogue. 
+
+Based on this, we might transform the LCSH documents to something like:
+
+```json
+{
+  "id": "<full LCSH URI>", // Necessary for indexing but not used externally
+  "identifier": {
+    "value": "sh12345678",
+    "identifierType": "lc-subjects",
+    "type": "Identifier"
+  },
+  "label": "<value of skos:prefLabel>",
+  "alternativeLabels": [
+    "<values of skos:altLabel>"
+  ]
+}
+```
+### Forwarding changes in external concepts to the concepts augmentation pipeline
+
+> Need to work this bit out
+
+
+
+## Questions
+
+- The format proposed above doesn't follow the existing schema for `identifierType` - rather than just the id we have previously used an object for `identifierType`, like
+  ```json
+  {
+    "id": "lc-subjects",
+    "label": "Library of Congress Subject Headings (LCSH)",
+    "type": "IdentifierType"
+  }
+  ```
+  Is this OK?
```

Comment:
Kind of depends what the docs are for. It’s possibly redundant data, though not exactly enough to be a problem, for internal service data, but something that we will want to add in the API.

Reply:
I think this is good.  There is plenty of opportunity to add in the redundant data later on, probably in the stage that populates the Index for the API.

The extra verbosity from storing the identifierType object all the way through isn't a problem for the computers, but if we are looking at it in logs or querying an interim DB for debugging, it adds clutter.  I'd rather just see it as a single property with the identifier for the type.